src/comment_provider/conversationProvider.ts

```diff
@@ -0,0 +1,575 @@
+
+      (thread.comments[0] as ConversationComment).conversation_id =
+        addReplyResult.conversation_id;
+      thread.share_id = shareId;
+      thread.conversation_group_id = addReplyResult.conversation_group_id;
+      thread.meta = meta;
+
+      this.dbtTerminal.debug(
+        "ConversationProvider",
+        "added conversation to created conversation group",
+        addReplyResult,
+      );
+
+      this._threads[shareId] = {
+        ...this._threads[shareId],
+        [thread.conversation_group_id]: thread,
+      };
+    } catch (error) {
+      this.dbtTerminal.error(
+        "ConversationProvider:createConversation",
+        "unable to create conversation",
+        error,
+      );
+      // If share cannot be created, delete the thread
+      reply.thread.dispose();
+      window.showErrorMessage(
+        extendErrorWithSupportLinks(
+          `Unable to save your comment. ${(error as Error).message}`,
+        ),
+      );
+    }
+  }
+
+  async replyToConversation(reply: CommentReply) {
+    const thread = reply.thread as ConversationCommentThread;
+    try {
+      if (!thread.share_id) {
+        throw new Error("Unable to find conversation. Missing share id");
+      }
+
+      this.dbtTerminal.debug(
+        "ConversationProvider:replyToConversation",
+        "replying to conversation",
+        reply,
+      );
+      this.addComment(reply);
+
+      if (!thread.conversation_group_id) {
+        if (!thread.share_id) {
+          throw new Error(
+            "Unable to find conversation. Missing conversation group id",
+          );
+        }
+        return;
+      }
+
+      await this.conversationService.addConversationToGroup(
+        thread.share_id,
+        thread.conversation_group_id,
+        this.convertTextToDbFormat(reply.text),
+      );
+    } catch (error) {
+      this.dbtTerminal.error(
+        "ConversationProvider:replyToConversation",
+        "unable to reply conversation",
+        error,
+      );
+      window.showErrorMessage(
+        extendErrorWithSupportLinks(
+          `Unable to save your reply. ${(error as Error).message}`,
+        ),
+      );
+    }
+  }
+
+  async resolveConversation(commentThread: ConversationCommentThread) {
+    try {
+      if (!commentThread.share_id) {
+        throw new Error("Unable to find conversation. Missing share id");
+      }
+      this.dbtTerminal.debug(
+        "ConversationProvider:resolveConversation",
+        `resolving conversation: ${commentThread.conversation_group_id} in share: ${commentThread.share_id}`,
+      );
+      const result = await this.conversationService.resolveConversation(
+        commentThread.share_id,
+        commentThread.conversation_group_id,
+      );
+      this.dbtTerminal.debug(
+        "ConversationProvider:resolveConversation",
+        `resolved conversation: ${commentThread.conversation_group_id} in share: ${commentThread.share_id}`,
+        result,
+      );
+
+      commentThread.dispose();
```

Comment:
Should we dispose when resolving?

Reply:
@anandgupta42  @mdesmet should we show the resolved conversations as well?

Reply:
I think it is a common action to unresolve a conversation.

Reply:
@mdesmet i had discussion with @anandgupta42 and we have decided not to show resolved conversations atleast in v1, since it could end up showing lots of comments and those resolved comments might not even in same context since more changes might have been added.